File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlAsExpr.cs
namespace T1.SqlSharp.Expressions;

public class SqlAsExpr : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.AsExpr;
    public TextSpan Span { get; set; } = new();

    public required ISqlExpression Instance { get; set; }
    public required ISqlExpression As { get; set; }

    public string ToSql()
    {
        return $"{Instance.ToSql()} as {As.ToSql()}";
    }
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_AsExpr(this);
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlAsExpr.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlAsExpr implements ISqlExpression {
    SqlType: SqlType = SqlType.AsExpr;
    Span: TextSpan = new TextSpan();
    Instance!: ISqlExpression;
    As!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_AsExpr(this);
    }

    ToSql(): string {
        return `${this.Instance.ToSql()} AS ${this.As.ToSql()}`;
    }
} 